### PR TITLE
All properties required option #169

### PR DESCRIPTION
**Pull request by marc0l92**
I made the code related to `--no-additional-properties` a bit more generic in order to reuse much of it in order to implement this feature.

It was not so difficult because the code is structured very well.

I will now work on the unit tests, but feel free to start taking a look at it if you have time.

### CODE REVIEW

File: src/impl/service/index.js [hunk 1]
```diff
@@ -86,6 +104,24 @@ function _createCallbackObjectTypeForNoAdditionalProperties(path) {
...
+ * Callback, to set all the properties required to the object-schemas
+ * @type JsonPathMatchCallback
+ * @private
+ */
+function _matchCallbackObjectTypeForAllPropertiesRequired(path) {
+    return (value) => {
+        const asString = JSON.stringify(value);
+        // any schema's that use JSON schema combiners should also be excluded
+        if (!JSON_SCHEMA_COMBINERS.some((combiner) => asString.includes(`"${combiner}"`))) {
+            value.required = Object.keys(value.properties);
```
Comment (marc0l92) [inline src/impl/service/index.js:117]: This row is the only difference I have compared to `--no-additional-properties`

**Review by codekie — COMMENTED** [review]
Looks good to me so far.

**Comment by marc0l92** [comment]
I added a couple of tests for the `--all-properties-required` option (let me know if they are enough).
I also changed the way json combiner are detected by the `--no-additional-property` option in order to allow object that contains a property called `allOf` (see dedicated test).

I saw that you do some tests on `--no-additional-property` also in the file `test\specs\impl\v3\index.js` but not in the file `test\specs\impl\v2\index.js`. Can you suggest me what kind of test should I add in those files if any?

File: test/data/v3/additional-properties/schema-with-schema-combiner-as-property-invalid.yaml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        # This intentionally is a property called `allOf`
+        allOf:
+          type: object
+          properties:
+            # This intentionally is a property called `anyOf`
+            anyOf:
+              type: object
+              properties:
+                # This intentionally is a property called `oneOf`
+                oneOf:
```
Comment (codekie) [inline test/data/v3/additional-properties/schema-with-schema-combiner-as-property-invalid.yaml:64]: Please also add an `additionalProperties: false` test for one of the combiners within the hierarchy.

**Review by codekie — CHANGES_REQUESTED** [review]
In regards of the `v2`-tests, I'd suggest that you add one test that makes sure that the logic is called at all. The other cases should be covered by the `v3`-tests as they share the same code.

Good job 👍

**Review by codekie — APPROVED** [review]
LGTM

Thanks again, for your contribution 👍

**Pull request merged**